Model/Service/SanitizingProductService.php

```diff
@@ -0,0 +1,85 @@
+ * are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice,
+ * this list of conditions and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following disclaimer in the documentation
+ * and/or other materials provided with the distribution.
+ *
+ * 3. Neither the name of the copyright holder nor the names of its contributors
+ * may be used to endorse or promote products derived from this software without
+ * specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+ * ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ * @author Nosto Solutions Ltd <contact@nosto.com>
+ * @copyright 2019 Nosto Solutions Ltd
+ * @license http://opensource.org/licenses/BSD-3-Clause BSD 3-Clause
+ *
+ */
+
+namespace Nosto\Tagging\Model\Service;
+
+use Exception;
+use Magento\Catalog\Api\Data\ProductInterface;
+use Nosto\Object\Product\Product;
+use Nosto\Tagging\Model\Service\ProductServiceInterface as NostoProductService;
+use Magento\Store\Api\Data\StoreInterface;
+use Nosto\Tagging\Logger\Logger as NostoLogger;
+
+class SanitizingProductService implements ProductServiceInterface
+{
+    /** @var ProductServiceInterface */
+    private $nostoProductService;
+    /**
+     * @var NostoLogger
+     */
+    private $nostoLogger;
+
+    /**
+     * DefaultProductService constructor.
+     * @param ProductServiceInterface $nostoProductService
+     * @param NostoLogger $nostoLogger
+     */
+    public function __construct(
+        NostoProductService $nostoProductService,
+        NostoLogger $nostoLogger
+    ) {
+        $this->nostoProductService = $nostoProductService;
+        $this->nostoLogger = $nostoLogger;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getProduct(ProductInterface $product, StoreInterface $store)
+    {
+        /** @var Product $nostoProduct */
+        $nostoProduct = $this->nostoProductService->getProduct(
+            $product,
+            $store
+        );
+        try {
+            return $nostoProduct->sanitize();
+        } catch (Exception $e) {
```

Comment:
Why? You're just logging it. Remove this.

Reply:
To avoid the crashing the website if something blows up during the sanitization. Guess we want that then?